rocAL/rocAL/include/pipeline/tensor.h

```diff
@@ -0,0 +1,263 @@
+/*
+Copyright (c) 2019 - 2022 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#pragma once
+
+#include <VX/vx.h>
+#include <VX/vx_types.h>
+#include <vector>
+#include <cstring>
+#include <array>
+#include <queue>
+#include <memory>
+#if ENABLE_HIP
+#include "hip/hip_runtime.h"
+#include "device_manager_hip.h"
+#else
+#include "device_manager.h"
+#endif
+#include "commons.h"
+
+/*! \brief Converts Rocal Memory type to OpenVX memory type
+ *
+ * @param mem input Rocal type
+ * @return the OpenVX type associated with input argument
+ */
+vx_enum vx_mem_type(RocalMemType mem);
```

Comment:
why is this function declared here? Please move common function definitions to .h file

Reply:
The vx_mem_type function is used only in tensor.cpp file. The image.cpp file will be removed in future once we replace with tensor pipeline hence we have declared the API in tensor.h and defined in tensor.cpp file